### PR TITLE
Adding test coverage for delayed readiness checks behavior

### PR DESCRIPTION
**Pull request by fabiobrz**
* Introducing `microprofile-manual-mode` *module* in order to run test cases that request for Wildfly container to be managed manually

* Test coverage for WFLY-12592:
Provided tests aim to verify the MP Health check implementation in complex scenarios where delayed readiness health checks could lead to misinterpretation of server/service status, see https://issues.redhat.com/browse/WFLY-12952.

* Jenkins job successful run: https://eap-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/eap-7.x-microprofile-simple-face/47 (only FT errors there)

@istraka FYI this is based on your initial reproducer at https://github.com/istraka/eap-microprofile-test-suite/blob/mp-health-feature/microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/delay/NotReadyHealthCheckTest.java

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Link to the passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

### CODE REVIEW

File: microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/NotReadyHealthCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+                .addClasses(DelayedLivenessHealthCheck.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                .as(ZipExporter.class)
+                .exportTo(deployment, true);
+
+        return deployment;
+    }
+
+    /**
+     * Unmanaged empty deployment to state that Arquillian shouldn't care about deployments at all
```
Comment (marekkopecky) [inline microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/NotReadyHealthCheckTest.java]: Is this method necessary? Can be removed?

File: microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/ReadinessChecker.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    private AtomicBoolean shouldStop = new AtomicBoolean(false);
+
+    private List<ReadinessState> states = new CopyOnWriteArrayList<>();
+
+    public static final String DEFAULT_READINESS_CHECK_NAME_PREFIX = "ready-deployment.";
+
+    @Override
+    public Boolean call() {
+        states.add(ReadinessState.START());
```
Comment (marekkopecky) [inline microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/ReadinessChecker.java]: What about to use `addState` to be consistent in this method? Same for `states.add(ReadinessState.END());`

File: microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/NotReadyHealthCheckTest.java (hunk 1 above)
Comment (fabiobrz) [reply microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/NotReadyHealthCheckTest.java]: Thanks, removing it.

File: microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/ReadinessChecker.java (hunk 2 above)
Comment (fabiobrz) [reply microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/ReadinessChecker.java]: Right, fixing.

**Comment by fabiobrz** [comment]
@marekkopecky, FYI: successful job run: https://eap-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/eap-7.x-microprofile-simple-face/47

**Pull request merged**